hub.go

```diff
@@ -331,8 +331,16 @@ func hubEntryToEntity(entry *hubEntry) *HubEntity {
 }
 
 // NewHub creates a new Event Hub client for sending and receiving messages
+// NOTE: If the AZURE_ENVIRONMENT variable is set, it will be used to set the ServiceBusEndpointSuffix
+// from the corresponding azure.Environment type at the end of the namespace host string. The default
+// is azure.PublicCloud.
 func NewHub(namespace, name string, tokenProvider auth.TokenProvider, opts ...HubOption) (*Hub, error) {
-	ns, err := newNamespace(namespaceWithAzureEnvironment(namespace, tokenProvider, azure.PublicCloud))
+	env, err := azure.EnvironmentFromName(os.Getenv("AZURE_ENVIRONMENT"))
```

Comment:
`azure.EnvironmentFromName()` assumes that the provided value isn't empty, so this will fail if the env var isn't set.  I think it would be best to default to `AzurePublicCloud` in that case.  It also means that if `err` isn't `nil` you'd return the error.